src/renderer/uia/UiaRenderer.cpp

```diff
@@ -47,6 +47,12 @@ UiaEngine::UiaEngine(IUiaEventDispatcher* dispatcher) :
 [[nodiscard]] HRESULT UiaEngine::Disable() noexcept
 {
     _isEnabled = false;
+
+    // If we had buffered any text from NotifyNewText, dump it. When we do come
+    // back around to actually paint, we will just no-op. No sense in keeping
+    // the data buffered.
+    _newOutput.clear();

```

Comment:
This will not release any memory however. Additionally, it also doesn't clear `_queuedOutput`. Do we need to do that?

This comment contains one approach to clear the memory in the `EndPaint()` method: https://github.com/microsoft/terminal/issues/16209#issuecomment-1773902806
In the `Disable()` case I'd release the memory whenever, let's say, `_newOutput.capacity() >= 4096` or something.

Reply:
Derp yea, we can resize it too. 

I didn't think we needed to clear out `_queuedOutput`, since we might technically be after `EndPaint`, but in the middle of a `Present`.

Reply:
Ah right, good point.